rkt/config/config.go

```diff
@@ -64,6 +65,76 @@ type Config struct {
 	Stage1                       Stage1Data
 }
 
+// MarshalJSON marshals the config for user output.
+func (c *Config) MarshalJSON() ([]byte, error) {
+	stage0 := []interface{}{}
+
+	for host, auth := range c.AuthPerHost {
+		var typ string
+		var credentials interface{}
+
+		switch h := auth.(type) {
+		case *basicAuthHeaderer:
+			typ = "basic"
+			credentials = h.auth
+		case *oAuthBearerTokenHeaderer:
+			typ = "oauth"
+			credentials = h.auth
+		default:
+			return nil, errors.New("unknown headerer type")
+		}
+
+		auth := struct {
+			RktVersion  string      `json:"rktVersion"`
+			RktKind     string      `json:"rktKind"`
+			Domains     []string    `json:"domains"`
+			Type        string      `json:"type"`
+			Credentials interface{} `json:"credentials"`
+		}{
+			RktVersion:  "v1",
+			RktKind:     "auth",
+			Domains:     []string{host},
+			Type:        typ,
+			Credentials: credentials,
+		}
+
+		stage0 = append(stage0, auth)
+	}
+
+	for registry, credentials := range c.DockerCredentialsPerRegistry {
+		dockerAuth := struct {
+			RktVersion  string           `json:"rktVersion"`
+			RktKind     string           `json:"rktKind"`
+			Registries  []string         `json:"registries"`
+			Credentials BasicCredentials `json:"credentials"`
+		}{
+			RktVersion:  "v1",
+			RktKind:     "dockerAuth",
+			Registries:  []string{registry},
+			Credentials: credentials,
+		}
+
+		stage0 = append(stage0, dockerAuth)
+	}
+
+	paths := struct {
+		RktVersion   string `json:"rktVersion"`
+		RktKind      string `json:"rktKind"`
+		Data         string `json:"data"`
+		Stage1Images string `json:"stage1-images"`
+	}{
+		RktVersion:   "v1",
+		RktKind:      "paths",
+		Data:         c.Paths.DataDir,
```

Comment:
@krnowak @alban this will print a different path if the user specified `--dir`, see https://github.com/coreos/rkt/blob/master/rkt/rkt.go#L261, what do you think, use the `calculateDataDir` result instead?